### PR TITLE
retrofitting performance improvement for getting all orgs, to master

### PR DESCRIPTION
**Pull request by ConorH994**
https://tools.hmcts.net/jira/browse/RDCC-899

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/CreateNewUserWithRolesTest.java [hunk 1]
```diff
@@ -446,32 +446,4 @@ public void validate_invalid_email_for_invite_user_and_throw_exception() {
         assertThat(newUserResponse.get("http_status")).isEqualTo("400");
 
     }
-
-    @Test
-    public void should_return_409_when_same_invited_new_user_is_super_user_of_pending_org() {
```
Comment (ConorH994) [inline src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/CreateNewUserWithRolesTest.java]: this test seemed to be removed in development branch so I am removing it here too

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java [hunk 2]
```diff
@@ -245,6 +215,47 @@ public OrganisationsDetailResponse retrieveOrganisations() {
...
+        List<Organisation> retrievedOrganisations = organisationRepository.findAll();
+
+        if (retrievedOrganisations.isEmpty()) {
+            throw new EmptyResultDataAccessException(1);
+        }
+
+        List<Organisation> pendingOrganisations = new ArrayList<>();
+        List<Organisation> activeOrganisations = new ArrayList<>();
+        List<Organisation> resultingOrganisations = new ArrayList<>();
+
```
Comment (KOTLAPRA23) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java]: activeOrganisations list is not required remove it and activeOrganisationDetails map size

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java [hunk 3]
```diff
@@ -245,6 +215,47 @@ public OrganisationsDetailResponse retrieveOrganisations() {
...
+                    activeOrganisationDetails.put(organisation.getUsers().get(0).getUserIdentifier(), organisation);
+                }
+            } else if (organisation.getStatus() == OrganisationStatus.PENDING) {
+                pendingOrganisations.add(organisation);
+            }
+        });
+
+        List<Organisation> updatedActiveOrganisations = new ArrayList<>();
+
+        if (!CollectionUtils.isEmpty(activeOrganisations)) {
```
Comment (KOTLAPRA23) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java]: activeOrganisationDetails size verify here in the if method

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java [hunk 4]
```diff
@@ -245,6 +215,47 @@ public OrganisationsDetailResponse retrieveOrganisations() {
...
+
+        List<Organisation> pendingOrganisations = new ArrayList<>();
+        List<Organisation> activeOrganisations = new ArrayList<>();
+        List<Organisation> resultingOrganisations = new ArrayList<>();
+
+        Map<String, Organisation> activeOrganisationDetails = new ConcurrentHashMap<>();
+
+        retrievedOrganisations.forEach(organisation -> {
+            if (organisation.isOrganisationStatusActive()) {
+                activeOrganisations.add(organisation);
```
Comment (KOTLAPRA23) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java]: remove activeOrganisations.add(organisation);

**Pull request merged**